### ISSUE TITLE
Loading excel from stream doesn't work with read-only file system

### ISSUE THREAD

**Issue by imperugo** [ClosedXML/ClosedXML#2445]
## Read and complete the full issue template

Do not randomly delete sections. They are here for a reason.

**Do you want to request a *feature* or report a *bug*?**
- [x] Bug
- [ ] Feature
- [ ] Question

**Did you test against the latest CI build?**
- [ ] Yes
- [x] No

If you answered `No`, please test with [the latest development build](https://ci.appveyor.com/project/ClosedXML/ClosedXML/branch/develop/artifacts) first.

**Version of ClosedXML**

e.g. 0.104.1

**What is the current behavior?**

Loading the worksheet from a stream (embedded resource in my case) I got an error because the pod file system is in read-only mode

Here the stack

```
System.IO.IOException: Read-only file system : '/tmp/'
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirError)
   at System.IO.Path.GetTempFileName()
   at DocumentFormat.OpenXml.Packaging.WriteableStreamExtensions.TemporaryFile..ctor()
   at DocumentFormat.OpenXml.Packaging.WriteableStreamExtensions.EnableWriteableStream(IFeatureCollection features)
   at DocumentFormat.OpenXml.Packaging.PackageUriHandlingExtensions.EnableUriHandling(IFeatureCollection features)
   at DocumentFormat.OpenXml.Packaging.PackageUriHandlingExtensions.EnableUriHandling(OpenXmlPackage package)
   at DocumentFormat.OpenXml.Builder.OpenXmlPackageBuilderExtensions.<>c__10`1.<UseDefaultBehaviorAndLockBuilder>b__10_0(TPackage package, PackageInitializerDelegate`1 next)
   at DocumentFormat.OpenXml.Builder.OpenXmlPackageBuilderExtensions.<>c__DisplayClass8_1`1.<Use>b__1(TPackage package)
   at DocumentFormat.OpenXml.Builder.OpenXmlPackageBuilder`1.Factory.Create(IPackageInitializer initializer)
   at DocumentFormat.OpenXml.Builder.OpenXmlPackageBuilderExtensions.Open[TPackage](IPackageFactory`1 builder, Stream stream, PackageOpenMode mode)
   at DocumentFormat.OpenXml.Builder.OpenXmlPackageBuilderExtensions.Open[TPackage](IPackageFactory`1 builder, Stream stream, Boolean isEditing)
   at DocumentFormat.OpenXml.Packaging.SpreadsheetDocument.Open(Stream stream, Boolean isEditable, OpenSettings openSettings)
   at DocumentFormat.OpenXml.Packaging.SpreadsheetDocument.Open(Stream stream, Boolean isEditable)
   at ClosedXML.Excel.XLWorkbook.LoadSheets(Stream stream)
   at ClosedXML.Excel.XLWorkbook.Load(Stream stream)
   at ClosedXML.Excel.XLWorkbook..ctor(Stream stream, LoadOptions loadOptions)
   at ClosedXML.Excel.XLWorkbook..ctor(Stream stream)
```

**What is the expected behavior or new feature?**

That it works without issue on read-only file system because I don't need to store anything on the storage

**Is this a regression from the previous version?**

Yes it is, because the previous version 0.102.3 works perfectly

## Reproducibility

Run this with read-only file system

```c#
public void Main()
{
      var fallbackFontStream = Assembly.GetExecutingAssembly().GetManifestResourceStream(templateId);
      var wb =  new XLWorkbook(fallbackFontStream);
}
```

**Comment by jahav** [comment]
I believe this bug should go to OpenXML SDK. This crashes directly on opening `SpreadsheetDocument.Open(stream, isEditable: false)`.

This is a change from 2 to 3, where they abstracted `System.IO.Package` to `IPackage` for various reasons (the believe the most important one was https://github.com/dotnet/runtime/issues/1544). 

There might be a way to deal with read only filesystem:
* Use a `new OpenSettings() { CompatibilityLevel = CompatibilityLevel.Version_2_20 }` but that would mean we miss out on various things a lot of people want in 3.0, like malformed URL not crashing everything.
* Use a ctor with old Package param (`System.IO.Packaging.Package.Open(path, FileMode.Open, FileAccess.Read)`) instead of stream. The feature that deals with copying to a file won't bet triggered then.

Neither of this appeal to me, because it is doing something  OpenXML SDK should do on it's own (i.e. detect read only filesystem and not use temp file), so please open a ticket in https://github.com/dotnet/Open-XML-SDK.

I am closing it for now. We can revisit, if OpenXML SDK doesn't consider it a bug.

**Comment by supermartzin** [comment]
Hi @jahav,

I've just encountered the same error when using your library **v0.105.0** and running my app on AKS cluster.
I tried the workaround you are mentioning above, I downloaded source code and changed the call of a constructor when opening the Excel file in LoadSheets method:

```
using (var dSpreadsheet = SpreadsheetDocument.Open(System.IO.Packaging.Package.Open(stream, FileMode.Open, FileAccess.Read)))
```

The problem is that I cannot make this change from outside the library while using it as a NuGet package.
Is it possible to introduce some special setting or new constructor so the file will be loaded with the old Package parameter?

Thanks a lot in advance!
